tests/unit/test_kwargs/test_multipart_data.py

```diff
@@ -292,6 +296,7 @@ def test_multipart_request_without_charset_for_filename() -> None:
         }
 
 
+@pytest.mark.xfail
```

Comment:
Figure out why this is failing. I suspect `filename*` might not be (fully) supported

Reply:
Yes, `filename*` is not supported because it is deprecated (actually forbidden) according to RFC-7578 section 4.2)

> NOTE: The encoding method described in [RFC5987], which would add a "filename*" parameter to the Content-Disposition header field, MUST NOT be used.

Reply:
Interesting, I was not aware of this. Hmm. We have previously supported this, so removing support for it now would constitute a breaking change. Do you think there's a reasonable workaround for us here while still using `multipart`?

Reply:
I'm open for adding optional features if there is an actual real-world use case for it, but `filename*` is a `MUST NOT` and I do not know any browser or HTTP client that actually does this. I'd argue that following the RFC and WHATWG HTML5 spec is a bugfix, not a breaking change ;) Breaking broken clients is okay.

Reply:
> Breaking broken clients should be allowed.

I agree on this.